ckanext/switzerland/harvester/base_sbb_harvester.py

```diff
@@ -825,89 +823,56 @@ def _import_stage(self, harvest_object):  # noqa
 
-                # take this metadata from the old version if available
-                resource_meta["rights"] = resource_meta.get("rights", "")
-                resource_meta["license"] = resource_meta.get("license", "")
-                resource_meta["coverage"] = resource_meta.get("coverage", "TODO")
-                resource_meta["description"] = resource_meta.get(
-                    "description", {"de": "", "en": "", "fr": "", "it": ""}
-                )
-                resource_meta["relations"] = resource_meta.get("relations", [])
+            resource_meta["identifier"] = os.path.basename(f)
 
-                log_msg = "Creating new resource: %s"
+            file_format, mimetype, mimetype_inner = self._get_mimetypes(f)
+            resource_meta["format"] = file_format
+            resource_meta["mimetype"] = mimetype
+            resource_meta["mimetype_inner"] = mimetype_inner
 
-            # -----------------------------------------------------
-            # create the resource, but use the known metadata (of the old resource)
-            # -----------------------------------------------------
-            else:
-                old_resource_id = resource_meta["id"]
+            resource_meta["name"] = {
+                "de": os.path.basename(f),
+                "en": os.path.basename(f),
+                "fr": os.path.basename(f),
+                "it": os.path.basename(f),
+            }
+            resource_meta["title"] = {
+                "de": os.path.basename(f),
+                "en": os.path.basename(f),
+                "fr": os.path.basename(f),
+                "it": os.path.basename(f),
+            }
 
-                self._reset_resource(resource_meta)
+            resource_meta["issued"] = now
+            resource_meta["version"] = now
 
-                log_msg = "Updating resource (with known metadata): %s"
+            # take this metadata from the old version if available
+            resource_meta["rights"] = old_resource_meta.get("rights", "")
+            resource_meta["license"] = old_resource_meta.get("license", "")
+            resource_meta["coverage"] = old_resource_meta.get("coverage", "TODO")
+            resource_meta["description"] = old_resource_meta.get(
+                "description", {"de": "", "en": "", "fr": "", "it": ""}
+            )
+            resource_meta["relations"] = old_resource_meta.get("relations", [])
+            resource_meta["rights"] = old_resource_meta.get("rights", [])
 
             resource_meta["package_id"] = dataset["id"]
 
             # url parameter is ignored for resource uploads, but required by ckan
             # this parameter will be replaced later by the resource patch with a link to
             # the download file
-            if "url" not in resource_meta:
-                resource_meta["url"] = "http://dummy-value"
-                resource_meta["download_url"] = None
+            resource_meta["url"] = "http://dummy-value"
```

Comment:
hmmm, maybe I am sensitive about this code snippet, but isn't this always setting dummy-value, even when it would happen, that the url-parameter is actually given?

Reply:
Totally reasonable to be sensitive to it at this point tbh.

I think we would never get the url parameter given, as we are only ever uploading files in these harvesters, never getting data about resources that are accessed by url. Previously, we would either get an old url from the old resource, which would be useless for the new resource, or we would use the dummy value. Now we just set the dummy value. The url is supposed to be set once the resource is uploaded anyway.

Does that sound right to you?